query.go

```diff
@@ -0,0 +1,114 @@
+package ensign
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	api "github.com/rotationalio/go-ensign/api/v1beta1"
+)
+
+// QueryCursor exposes event results from an EnSQL query with familiar database cursor
+// semantics.
+type QueryCursor struct {
+	stream api.Ensign_EnSQLClient
+}
```

Comment:
Notable that there is no mutex on the stream -- we probably want to update the comment to tell the user that this is not thread-safe. 

```suggestion
// QueryCursor exposes event results from an EnSQL query with familiar database cursor
// semantics. Note that the cursor is not thread safe and should only be read from a single
// thread to ensure consistency without race conditions.
type QueryCursor struct {
	stream api.Ensign_EnSQLClient
}
```